File: 1431. Kids with the greatest number of candies/main.cpp
#include <vector>
using namespace std;
class Solution
{
public:
    vector<bool> kidsWithCandies(vector<int> &candies, int extraCandies)
    {
        vector<bool> result;
        bool bool_max;
        // find max value
        int max_value = 0;
        for (int x : candies)
        {
            if (x >= max_value)
            {
                max_value = x;
            }
        }

        // determine whether kids have the greatest number of canides or not
        for (int candy : candies)
        {
            if (candy + extraCandies >= max_value)
            {
                bool_max = true;
            }
            else
            {
                bool_max = false;
            }
            result.push_back(bool_max);
        }
        return result;
    }
};

File: 1431. Kids with the greatest number of candies/main.py
class Solution(object):
    def kidsWithCandies(self, candies, extraCandies):
        """
        :type candies: List[int]
        :type extraCandies: int
        :rtype: List[bool]
        """
        #find the gratest value
        max_value=0
        results=[]
        for candy in candies:
            if(candy>max_value):
                max_value=candy
        
        #determine whether kids have the greatest number of canides or not
        for candy in candies:
            if(candy+extraCandies>=max_value):
               results.append(True)
            else:
                results.append(False)
        return results
